pkg/cmd/registry/rule/list/list.go

```diff
@@ -0,0 +1,193 @@
+	"github.com/spf13/cobra"
+
+	registryinstanceclient "github.com/redhat-developer/app-services-sdk-go/registryinstance/apiv1internal/client"
+)
+
+const (
+	ruleValidity      = "validity"
+	ruleCompatibility = "compatibility"
+)
+
+const (
+	ruleDisabled = "disabled"
+	ruleEnabled  = "enabled"
+)
+
+// ruleRow is the details of a Service Registry rules needed to print to a table
+type ruleRow struct {
+	RuleType string `json:"ruleType" header:"Rule Type"`
+
+	Description string `json:"description,omitempty" header:"Description"`
+
+	Status string `json:"status" header:"Status"`
+}
+
+type options struct {
+	IO         *iostreams.IOStreams
+	Config     config.IConfig
+	Connection factory.ConnectionFunc
+	Logger     logging.Logger
+	localizer  localize.Localizer
+	Context    context.Context
+
+	registryID string
+	artifactID string
+	group      string
+}
+
+// NewListCommand creates a new command to view status of rules
+func NewListCommand(f *factory.Factory) *cobra.Command {
+
+	opts := &options{
+		IO:         f.IOStreams,
+		Config:     f.Config,
+		Connection: f.Connection,
+		Logger:     f.Logger,
+		localizer:  f.Localizer,
+		Context:    f.Context,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   f.Localizer.MustLocalize("registry.rule.list.cmd.description.short"),
+		Long:    f.Localizer.MustLocalize("registry.rule.list.cmd.description.long"),
+		Example: f.Localizer.MustLocalize("registry.rule.list.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) (err error) {
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			instanceID, ok := cfg.GetServiceRegistryIdOk()
+			if !ok {
+				return opts.localizer.MustLocalizeError("artifact.cmd.common.error.noServiceRegistrySelected")
+			}
+
+			opts.registryID = instanceID
+
+			return runList(opts)
+		},
+	}
+
+	flags := rulecmdutil.NewFlagSet(cmd, f)
+
+	flags.AddRegistryInstance(&opts.registryID)
+
+	flags.AddArtifactID(&opts.artifactID)
+	flags.AddGroup(&opts.group)
+
+	return cmd
+
+}
+
+func runList(opts *options) error {
+
+	var httpRes *http.Response
+	var newErr error
+	var enabledRules []registryinstanceclient.RuleType
+
+	ruleErrHandler := &rulecmdutil.RuleErrHandler{
+		Localizer: opts.localizer,
+	}
+
+	conn, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	dataAPI, _, err := conn.API().ServiceRegistryInstance(opts.registryID)
+	if err != nil {
+		return err
+	}
+
+	if opts.artifactID == "" {
+		s := spinner.New(opts.IO.ErrOut, opts.localizer)
+		s.SetLocalizedSuffix("registry.rule.list.log.info.fetching.globalRules")
+		s.Start()
+
+		req := dataAPI.AdminApi.ListGlobalRules(opts.Context)
+
+		enabledRules, httpRes, newErr = req.Execute()
+		if httpRes != nil {
+			defer httpRes.Body.Close()
+		}
+
+		s.Stop()
+	} else {
+		s := spinner.New(opts.IO.ErrOut, opts.localizer)
+		s.SetLocalizedSuffix("registry.rule.list.log.info.fetching.artifactRules")
```

Comment:
"Fetching global rules for artifacts in Service Registry Instance" disappears once the data has loaded Do you want that? Maybe it is better to not have any message at all since I am not sure if it tells the user something they don't know.

Reply:
Otherwise it would be better to let that message remain on the top of the display even after it has completed.

Reply:
Same goes in other instances where this happens.

Reply:
> Same goes in other instances where this happens.

Is this suggestion for list command only or all commands that use spinners?
I cant form a strong opinion about the loader text. We can probably remove it as it shows nothing new.
Cc @wtrocki

Reply:
The loader text is fine but it should not just flash up for a half second and disappear - the user will be thinking they missed something important. It is best to ensure it just does not be deleted so the user can still see the message as normal.

Reply:
Sounds good. Updating PR to keep the spinners.

Reply:
Persisting the spinner texts runs to newline issue, `logger.info()` or `\n` for spacing doesn't seem to fix this either.
```
⣷ Fetching rules for the artifact❌ Artifact with ID 41590799-c99f-47cc-85a2-c50578 not found. Run the command in verbose mode using the -v flag to see more information
```
Any instance where we have leave spinner unclosed?